### ISSUE TITLE
APIBuilder should generate function pages

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3241]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3241|
|Type|task|
|Created|2008-05-18 20:28:49Z|
|Branch|https://github.com/twisted/twisted/tree/apibuilder-html-function-pages-3241|

The howto docs generate links to functions.  `APIBuilder` doesn't currently pass the pydoctor option necessary for these links to work, though.  The option is  `--html-write-function-pages`.

<details><summary>Searchable metadata</summary>

```
trac-id__3241 3241
type__task task
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_apibuilder_html_function_pages_3241 branches/apibuilder-html-function-pages-3241
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1211142529000000 1211142529000000
changetime__1231172068000000 1231172068000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

It does this now, somehow.  I forget how.  Anyway, eg http://twistedmatrix.com/projects/core/documentation/howto/pb-usage.html links to http://twistedmatrix.com/documents/8.1.0/api/twisted.spread.pb.RemoteReference.callRemote.html and the link works, so this is resolved.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `reopened`|
|-|-|

Someone must have updated the website without using APIBuilder in order to get those function pages.  APIBuilder itself will not generate them.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [25752]) Branching to 'apibuilder-html-function-pages-3241'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [25783]) Merge apibuilder-html-function-pages-3241

Author: exarkun
Reviewer: therve
Fixes: [#3241](https://github.com/twisted/twisted/issues/3241)

Pass the `--html-write-function-pages` option when invoking pydoctor to generate
API documentation.  This ensures that pages exist for functions and methods so that
other documents can link to these pages.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
